xbmc/pvr/timers/PVRTimerInfoTag.cpp

```diff
@@ -328,6 +317,61 @@ std::string CPVRTimerInfoTag::GetStatus() const
   return strReturn;
 }
 
+/**
+ * Get the type string of this Timer
+ */
+std::string CPVRTimerInfoTag::GetType() const
+{
+  std::string strReturn = g_localizeStrings.Get(231);
+  CSingleLock lock(m_critSection);
+
+  switch (m_iTimerType)
+  {
+  case PVR_TIMERTYPE_MANUAL_ONCE:
+  case PVR_TIMERTYPE_EPG_ONCE:
+    strReturn = g_localizeStrings.Get(805);
+    break;
+  case PVR_TIMERTYPE_MANUAL_SERIE:
+    strReturn = GetWeekdayString();
+    break;
+  default:
+    {
+      int strId = g_PVRClients->GetTimerTypeLocalizedStringId(m_iTimerType, m_iClientId);
+      strReturn = strId >= 0 ? g_PVRClients->GetClientLocalizedString(m_iClientId, strId) : g_localizeStrings.Get(812);
+      break;
+    }
+  }
+
+  return strReturn;
+}
+
+std::string CPVRTimerInfoTag::GetWeekdayString() const
+{
+  std::string strReturn = g_localizeStrings.Get(231);
+  CSingleLock lock(m_critSection);
+
+  if (m_iWeekdays == BITFLAG_MONDAY)         strReturn = g_localizeStrings.Get(813);
+  else if (m_iWeekdays == BITFLAG_TUESDAY)   strReturn = g_localizeStrings.Get(814);
+  else if (m_iWeekdays == BITFLAG_WEDNESDAY) strReturn = g_localizeStrings.Get(815);
+  else if (m_iWeekdays == BITFLAG_THURSDAY)  strReturn = g_localizeStrings.Get(816);
+  else if (m_iWeekdays == BITFLAG_FRIDAY)    strReturn = g_localizeStrings.Get(817);
+  else if (m_iWeekdays == BITFLAG_SATURDAY)  strReturn = g_localizeStrings.Get(818);
+  else if (m_iWeekdays == BITFLAG_SUNDAY)    strReturn = g_localizeStrings.Get(819);
+  else if (m_iWeekdays == BITFLAG_WEEKDAYS)  strReturn = g_localizeStrings.Get(810);
+  else if (m_iWeekdays == BITFLAG_WEEKENDS)  strReturn = g_localizeStrings.Get(811);
+  else if (m_iWeekdays == BITFLAG_ALL_DAYS)  strReturn = g_localizeStrings.Get(820);
+  else strReturn = StringUtils::Format("%s-%s-%s-%s-%s-%s-%s",
+      m_iWeekdays & BITFLAG_MONDAY    ? g_localizeStrings.Get(19149).c_str() : "__",
+      m_iWeekdays & BITFLAG_TUESDAY   ? g_localizeStrings.Get(19150).c_str() : "__",
+      m_iWeekdays & BITFLAG_WEDNESDAY ? g_localizeStrings.Get(19151).c_str() : "__",
+      m_iWeekdays & BITFLAG_THURSDAY  ? g_localizeStrings.Get(19152).c_str() : "__",
+      m_iWeekdays & BITFLAG_FRIDAY    ? g_localizeStrings.Get(19153).c_str() : "__",
+      m_iWeekdays & BITFLAG_SATURDAY  ? g_localizeStrings.Get(19154).c_str() : "__",
+      m_iWeekdays & BITFLAG_SUNDAY    ? g_localizeStrings.Get(19155).c_str() : "__");
+
+  return strReturn;
```

Comment:
Well, this is how it was done before, so I leaved it like that..
I'm now only using this for the special cases (where multiple bitflags are used)